File: old-regressions/cs/tst3.cs

/*++
Copyright (c) 2015 Microsoft Corporation

--*/

using Microsoft.Z3;
using System;
using System.Collections.Generic;

class Test
{

    public void Run()
    {
        using (Context ctx1 = new Context(new Dictionary<string, string>() { { ":relevancy", "0" } }))
        using (Context ctx2 = new Context(new Dictionary<string, string>() { { ":model", "false" },
                                                                             { ":pp-decimal", "true" },
                                                                             { ":relevancy", "2" },
                                                                             { ":pp-decimal-precision", "50" } }))
        {
            IntExpr x = ctx1.MkIntConst("x");
            IntExpr _x = (IntExpr) x.Translate(ctx2);

            Console.WriteLine(ctx2.MkEq(_x, ctx1.MkAdd(x, ctx1.MkInt(1)).Translate(ctx2)));
            Console.WriteLine(ctx2.MkPower(ctx2.MkReal(2), ctx2.MkReal(1, 2)).Simplify());
            
            Console.WriteLine(ctx1.MkPower(ctx1.MkReal(2), ctx1.MkReal(1, 2)).Simplify());

            Solver s = ctx1.MkSolver();
            s.Assert(ctx1.MkEq(x, ctx1.MkInt(2)));
            Console.WriteLine(s);
            Console.WriteLine(s.Check());
            Console.WriteLine(s.Model);
            
            s = ctx2.MkSolver();
            s.Assert(ctx2.MkEq(_x, ctx2.MkInt(2)));
            Console.WriteLine(s.Check());

            try
            {
                Console.WriteLine(s.Model);
            }
            catch (Z3Exception ex)
            {
                Console.WriteLine("failed: " + ex.Message);
            }
        } 
    }
}

File: old-regressions/python/tst3.py

# Copyright (c) 2015 Microsoft Corporation
from z3 import *

ctx1 = Context(relevancy=0)
ctx2 = Context(':model', False, ':pp-decimal', True, relevancy=2, pp_decimal_precision=50)
x = Int('x', ctx1)
_x = x.translate(ctx2)
print(_x == (x + 1).translate(ctx2))
print(simplify(Sqrt(2, ctx2)))
# pp_params is a global variable :-(
print(simplify(Sqrt(2, ctx1)))
s = Solver(ctx=ctx1)
s.add(x == 2)
print(s)
print(s.check())
print(s.model())
s = Solver(ctx=ctx2)
s.add(_x == 2)
print(s.check())
try:
    print(s.model())
except Z3Exception as ex:
    print("failed: %s" % ex)
